### PR TITLE
tested and working copy, has been running stable for a few years now

### PR DESCRIPTION
**Pull request by dasilva333**
tested and working with waterline 0.10.9

### CODE REVIEW

File: lib/query.js [hunk 1]
```diff
@@ -175,7 +189,9 @@ Query.prototype._build = function(criteria) {
 Query.prototype.where = function(options) {
   var self = this,
       operators = this.operators();
-
+  
+  //console.log("options " + JSON.stringify(options));
```
Comment (AndrewJo) [inline lib/query.js]: Can you remove this line?

File: lib/query.js [hunk 2]
```diff
@@ -63,10 +78,9 @@ Query.prototype.update = function(table, criteria, data) {
   str = str.slice(0, -2);
 
   this._query += 'SET ' + str + ' ';
-
+
```
Comment (AndrewJo) [inline lib/query.js]: Remove trailing whitespace.

File: lib/query.js [hunk 3]
```diff
@@ -30,7 +35,17 @@ var Query = function(schema) {
...
+  // Normalize joins key, allows use of both join and joins
+  if (criteria.join) {
+    criteria.joins = _.cloneDeep(criteria.join);
+    delete criteria.join;
+  }
+  //console.log("query.find")
+  //console.log(arguments);
+  //console.log(this);
+//  console.log(abort0)
```
Comment (AndrewJo) [inline lib/query.js]: Remove unnecessary commented code.

File: lib/query.js [hunk 4]
```diff
@@ -15,13 +16,17 @@ var _ = require('underscore'),
...
+	//console.log(arguments);
+	
   this._values = [];
   this._paramCount = 1;
   this._query = '';
+  this._tableDefs = tableDefs || {};
 
   this._schema = _.clone(schema);
-
+
```
Comment (AndrewJo) [inline lib/query.js]: Remove trailing whitespace.

File: lib/query.js [hunk 5]
```diff
@@ -15,13 +16,17 @@ var _ = require('underscore'),
  * If you have any questions, contact Andrew Jo <andrewjo@gmail.com>
  */
 
-var Query = function(schema) {
+var Query = function(schema, tableDefs) {
+	//console.log("new query");
+	//console.log(arguments);
```
Comment (AndrewJo) [inline lib/query.js]: Remove unnecessary commented code.

File: lib/adapter.js [hunk 6]
```diff
@@ -205,57 +169,147 @@ module.exports = (function() {
...
 
-        // Run the query
+        // Check for any index attributes
+        var indices = utils.buildIndexes(definition);
+
+        // Build query
+        var query = 'CREATE TABLE ' + table + ' (' + _schema + ')';
+        //client.on("trace", console.log)
+        //client.on("profile", console.log)
+        //console.log(client.run);
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 7]
```diff
@@ -205,57 +169,147 @@ module.exports = (function() {
...
+              //console.log(normalizedSchema);
+              try {
+            	// Set internal schema mapping
+                  collection.schema = normalizedSchema;
+            	  
+              } catch(e){
+            	  console.log(e);
+            	  //console.log(connection.collections[table]);
+            	  //console.log(table)
+            	  //console.log(connection)
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 8]
```diff
@@ -205,57 +169,147 @@ module.exports = (function() {
...
                 return cb(err);
               }
-
+              //console.log("schema")
+              //console.log(schema)
               var normalizedSchema = utils.normalizeSchema(schema);
-
-              // Set internal schema mapping
-              dbs[table].schema = normalizedSchema;
+              //console.log(normalizedSchema);
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 9]
```diff
@@ -170,9 +130,10 @@ module.exports = (function() {
           var index = { columns: [] };
 
           client.each(indexListQuery, function(err, currentIndex) {
-
+        	  //console.log(currentIndex)
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 10]
```diff
@@ -23,142 +25,100 @@ module.exports = (function() {
...
+        // See: http://www.sqlite.org/faq.html#q7)
         var query = 'SELECT * FROM sqlite_master WHERE type="table" AND name="' + table + '" ORDER BY name';
-
+        
         client.get(query, function (err, schema) {
           if (err || !schema) return cb();
-
-          // Query to get information about each table (see http://www.sqlite.org/pragma.html#pragma_table_info)
+          //console.log("client.get")
+          //console.log(schema);
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 11]
```diff
@@ -23,142 +25,100 @@ module.exports = (function() {
...
+     * is killed. Useful for tearing-down remaining open connections,
+     * etc.
+     *
+     * @param  {[type]}   connectionId [description]
+     * @param  {Function} cb           [description]
+     * @return {[type]}                [description]
+     */
+    teardown: function(connectionId, cb) {
+      if (!connections[connectionId]) return cb();
+      //console.log("Tearing down connection " + connectionId)
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 12]
```diff
@@ -23,142 +25,100 @@ module.exports = (function() {
...
+     * at server-start-time.  This is the only required method.
+     *
+     * @param  {[type]}   connection [description]
+     * @param  {[type]}   collection [description]
+     * @param  {Function} cb         [description]
+     * @return {[type]}              [description]
+     */
+    registerConnection: function(connection, collections, cb) {
+    	//console.log("registering connection for " + connection.identity);
+    	//console.log(cb.toString())
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 13]
```diff
@@ -360,110 +448,156 @@ module.exports = (function() {
...
+    	  // Build a query for the specific query strategy
+    	  var _query = new Query(_schema);
+    	  var query = _query.destroy(table, options);
+    	  
+    	  // Run Query
+    	  adapter.find(connectionId, table.replace(/["']/g, ""), options, function(err, models) {
+    		  //if (err) { console.log(err); return cb(err); }
+    		  
+    		  //console.log("adapter.find")
+    		  //console.log(arguments)
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 14]
```diff
@@ -360,110 +448,156 @@ module.exports = (function() {
...
-          });
-        });
-      }, dbs[table].config, cb);
+    destroy: function(connectionId, table, options, cb) {
+    	
+      spawnConnection(connectionId, function __DELETE__(client, cb) {
+
+    	  var connection = connections[connectionId];
+    	  var collection = connection.collections[table];
+    	  //console.log("definition: "  + JSON.stringify(collection.definition))
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/adapter.js [hunk 15]
```diff
@@ -574,26 +704,47 @@ module.exports = (function() {
...
+	  Object.keys(collections).forEach(function(collectionId) {
+      	schema[collectionId] = collections[collectionId].schema;
+      });
+	  return schema; 
+  }
+  
+  function spawnConnection(connectionName, logic, cb) {
+	  //console.log("spawnConnection")
+	  //console.log("connectionName " + connectionName)
+	  //console.log(connections)
```
Comment (AndrewJo) [inline lib/adapter.js]: Remove commented code.

File: lib/utils.js [hunk 16]
```diff
@@ -45,74 +45,120 @@ utils.buildSchema = function(obj) {
...
 
 /**
  * Builds a Select statement determining if Aggeragate options are needed.
  */
 
-utils.buildSelectStatement = function(criteria, table) {
+utils.buildSelectStatement = function(criteria, table, attributes, schema) {
+	//console.log("buildSelectStatement")
+	//console.log(arguments);
+	//abort0;
```
Comment (AndrewJo) [inline lib/utils.js]: Remove commented code.

File: lib/utils.js [hunk 17]
```diff
@@ -123,8 +169,82 @@ utils.buildSelectStatement = function(criteria, table) {
...
 
-  // Else select ALL
-  return 'SELECT rowid, * FROM ' + table + ' ';
+
+  query += 'SELECT rowid, ';
+
+  var selectKeys = [], joinSelectKeys = [];
+  //console.log("table: " + table);
+  //console.log("schema " + JSON.stringify(schema));
+  //console.log("table", table, "schema", schema);
```
Comment (AndrewJo) [inline lib/utils.js]: Remove commented code.

**Comment by AndrewJo** [comment]
@dasilva333 Thank you so much for the PR!! I reviewed your changes and made comments. Once they're addressed, I'll merge it in.

**Comment by dasilva333** [comment]
@AndrewJo sorry for the delay, the latest push to master has the code cleaned up.
Thanks

**Pull request merged**